src/Intl/Grapheme/Grapheme.php

```diff
@@ -123,7 +123,7 @@ public static function grapheme_substr($s, $start, $len = null)
             return false;
         }
         if ($start >= $slen) {
-            return false;
+            return \PHP_VERSION_ID >= 80000 ? '' : false;
```

Comment:
what about L123 & L138?

Reply:
Thanks a lot for the review @nicolas-grekas, you are right, those two `false` statements need to be changed to conditionally return empty strings on PHP 8 too. 

There was not enough test coverage for those edge cases, so the new commit extends the tests with them too: 
 - `grapheme_substr($c, -42, 0)`
 - `grapheme_substr($c, 42, 5)`